### ISSUE TITLE
Pass a property to a component

### ISSUE THREAD

**Issue by pmcelhaney** [Simplr/simplr-router#54]
In addition to the properties that come from the URL, I would like to "configure" the component by passing in additional properties. 

```js
import api from './my-api.js';

const routes = [
    { path: "", component: "simplr-router-demo" },
    { path: "example", component: "example-view", properties: { api } },
];
```

Why? Following the Dependency Inversion Principle, I don't want `<example-view>` to depend directly on a specific implementation of `api`. Instead, the `api` property is set by the host code (e.g. the router or a unit test). 

If you're okay with it, I'm happy to open a pull request.

**Comment by Matsuuu** [comment]
I've discussed this matter before in Slack, and in that discussion I argued against it since there was no good example of the case. 

You however have now changed my mind with the Dependency Inversion Principle. I think that could be a good addition to the router, and it actually is quite a minimal change that needs to be done to have it work.

Thinking at it, it could also be used to pass stuff like localization information too, and that's a use case I can get behind.

I'll put this on my worklist since I need to test a couple of other things related to this too, and will report back to you as I progress with this
